tedana/io.py

```diff
@@ -578,16 +589,18 @@ def write_split_ts(data, mmix, mask, comptable, io_generator, echo=0):
     if len(acc) != 0:
         if echo != 0:
             fout = io_generator.save_file(hikts, "high kappa ts split img", echo=echo)
-        else:
+            LGR.info(f"Writing high-Kappa time series: {fout}")
+        elif io_generator.verbose:
             fout = io_generator.save_file(hikts, "high kappa ts img")
-        LGR.info(f"Writing high-Kappa time series: {fout}")
+            LGR.info(f"Writing high-Kappa time series: {fout}")
```

Comment:
Unnecessary duplication here, if I'm not mistaken.

Reply:
I think this is correct, but confusing. There's a comment in the doc string, but I'll add more comments here to clarify. `write_split_ts` is called from two places:
In `tedana.py` there's a call to `io.writeresults_echoes` that is used to write out the time series for each echo separate and that only happens when the verbose flag is used (but others could use this function to write out echo time series without the verbose flag). That triggers the `LGR.info` statement on line 592. https://github.com/handwerkerd/tedana/blob/4727198f2ef7ef8a095d1e00081d2f0231f8065e/tedana/workflows/tedana.py#L819

The second is in `writeresults` which is used to write the denoised time series and, if `verbose` then also write out the high kappa and low kappa time series. https://github.com/handwerkerd/tedana/blob/4727198f2ef7ef8a095d1e00081d2f0231f8065e/tedana/io.py#L655

As written, if it's NOT verbose, then neither the `if` nor `elif` statements will be triggered so we don't want that `LGR.info` line separately executed.

Reply:
Ah, my bad. I missed that it was changed from else to elif. Sorry for the noise!

Reply:
I made a fix to the docstring and added a comment here because this section of code confused me (and you) every time I've looked it it.